lib/validators/specResolver.ts

```diff
@@ -167,85 +121,83 @@ export class SpecResolver {
     this.specInJson = specInJson
     this.specPath = specPath
     this.specDir = path.dirname(this.specPath)
-    this.visitedEntities = {}
-    this.resolvedAllOfModels = {}
-    if (!options) { options = {} }
-    if (options.shouldResolveRelativePaths === null
-      || options.shouldResolveRelativePaths === undefined) {
-      options.shouldResolveRelativePaths = true
-    }
-    if (options.shouldResolveXmsExamples === null
-      || options.shouldResolveXmsExamples === undefined) {
-      options.shouldResolveXmsExamples = true
-    }
+
+    options = defaultIfUndefinedOrNull(options, {})
+
+    options.shouldResolveRelativePaths = defaultIfUndefinedOrNull(
+      options.shouldResolveRelativePaths, true)
+
+    options.shouldResolveXmsExamples = defaultIfUndefinedOrNull(
+      options.shouldResolveXmsExamples, true)
+
     if (options.shouldResolveAllOf === null || options.shouldResolveAllOf === undefined) {
       if (!_.isUndefined(specInJson.definitions)) {
         options.shouldResolveAllOf = true
       }
```

Comment:
cant we pass this with a ternary operator to `defaultUndefinedOrNull` ?

Reply:
Not really, the logic here is slightly different.